server/src/routes/users.ts

```diff
@@ -97,51 +89,62 @@ router.post(
 	})
 );
 
+//get profile page
+router.get(
+	'/:id/profile',
+	sessionExtractor,
+	asyncHandler(async (req: CustomRequest, res) => {
+		if (!req.session || !req.session.userId || req.session.userId !== req.params.id) {
```

Comment:
last check in if is not necessary, as public profiles are extracted from this endpoint